File: Src/Icm.ContextConsole/DependencyResolver/IDependencyResolver.vb
''' <summary>
''' Dependency resolvers
''' </summary>
''' <remarks></remarks>
Public Interface IDependencyResolver

    ''' <summary>
    ''' Resolves a type by giving an instance of that type.
    ''' </summary>
    ''' <param name="service"></param>
    ''' <returns></returns>
    ''' <remarks></remarks>
    Function GetService(service As Type) As Object

    Sub ClearRequestScope()

End Interface

File: Src/Icm.ContextConsole/DependencyResolver/IDependencyResolver.cs
using System;

/// <summary>
/// Dependency resolvers
/// </summary>
/// <remarks></remarks>
public interface IDependencyResolver
{

	/// <summary>
	/// Resolves a type by giving an instance of that type.
	/// </summary>
	/// <param name="service"></param>
	/// <returns></returns>
	/// <remarks></remarks>
	object GetService(Type service);


	void ClearRequestScope();
}

//=======================================================
//Service provided by Telerik (www.telerik.com)
//Conversion powered by NRefactory.
//Twitter: @telerik
//Facebook: facebook.com/telerik
//=======================================================
